mailbox/noise.go

```diff
@@ -298,19 +303,38 @@ type handshakeState struct {
 
 	remoteStatic    *btcec.PublicKey // nolint
 	remoteEphemeral *btcec.PublicKey // nolint
+
+	password        []byte
+	payloadToSend   []byte
+	receivedPayload []byte
+
+	pattern HandshakePattern
+
+	version byte
 }
 
 // newHandshakeState returns a new instance of the handshake state initialized
-// with the prologue and protocol name. If this is the responder's handshake
-// state, then the remotePub can be nil.
-func newHandshakeState(initiator bool, prologue []byte,
-	localPub keychain.SingleKeyECDH) handshakeState {
+// with the prologue and protocol name.
+func newHandshakeState(version byte, pattern HandshakePattern, initiator bool,
+	prologue []byte, localPub keychain.SingleKeyECDH,
+	remoteStatic *btcec.PublicKey, password []byte,
+	authData []byte) (handshakeState, error) {
 
 	h := handshakeState{
-		initiator:   initiator,
-		localStatic: localPub,
+		version:        version,
+		symmetricState: symmetricState{},
+		initiator:      initiator,
+		localStatic:    localPub,
+		remoteStatic:   remoteStatic,
+		pattern:        pattern,
+		password:       password,
+		payloadToSend:  authData,
 	}
 
+	protocolName := fmt.Sprintf(
+		"Noise_%s_%s_%s_%s", pattern.Name, dhFn, cipherFn, hashFn,
```

Comment:
Need to make sure we're not making a backwards incompatible change here, as this is used to generate the very first handshake digest.